### PR TITLE
[SDA-8354] Allow creating operator roles prior to cluster spec

### PR DESCRIPTION
**Pull request by gdbranco**
Related issue: https://issues.redhat.com/browse/SDA-8354
# What
Allow creating operator roles prior to cluster spec

# Why
Ensures full BYO OIDC flow creating all needed resources prior to cluster spec

### CODE REVIEW

File: cmd/create/operatorroles/cmd.go [hunk 1]
```diff
@@ -197,40 +208,253 @@ func run(cmd *cobra.Command, argv []string) error {
...
+	if len(operatorRolesPrefix) > 32 {
+		r.Reporter.Errorf("Expected a prefix with no more than 32 characters")
+		os.Exit(1)
+	}
+	if !aws.RoleNameRE.MatchString(operatorRolesPrefix) {
+		r.Reporter.Errorf("Expected valid operator roles prefix matching %s", aws.RoleNameRE.String())
+		os.Exit(1)
+	}
+	oidcEndpointUrl := args.oidcEndpointUrl
+	if oidcEndpointUrl == "" {
```
Comment (oriAdler) [inline cmd/create/operatorroles/cmd.go:250]: We can check all the nessacery flags prior to this function call, and fail the flow earlier.

File: cmd/create/operatorroles/cmd.go [hunk 2]
```diff
@@ -197,40 +208,253 @@ func run(cmd *cobra.Command, argv []string) error {
...
-		r.Reporter.Errorf("Expected a valid path for '%s': %v", cluster.AWS().STS().RoleARN(), err)
+		r.Reporter.Errorf("%s", err)
+		os.Exit(1)
+	}
+	if parsedURI.Scheme != helper.ProtocolHttps {
+		r.Reporter.Errorf("Expected OIDC endpoint URL '%s' to use an https:// scheme", oidcEndpointUrl)
+		os.Exit(1)
+	}
+	installerRoleArn := args.installerRoleArn
+	if installerRoleArn == "" {
```
Comment (oriAdler) [inline cmd/create/operatorroles/cmd.go:264]: Same here, we can check the flags at the beginning of the command.
If the user provided `PrefixFlag`, we can check they provided the other required flags for BYO OIDC.

File: cmd/create/operatorroles/cmd.go [hunk 3]
```diff
@@ -197,40 +208,253 @@ func run(cmd *cobra.Command, argv []string) error {
...
-	path, err := getPathFromInstallerRole(cluster)
+	if args.prefix != "" {
+		return handleOperatorRoleCreationByPrefix(r, env, permissionsBoundary,
+			mode, policies, defaultPolicyVersion)
+	}
+	return handleOperatorRoleCreationByClusterKey(r, env, permissionsBoundary,
+		mode, policies, defaultPolicyVersion)
+}
+
+func handleOperatorRoleCreationByPrefix(r *rosa.Runtime, env string,
```
Comment (oriAdler) [inline cmd/create/operatorroles/cmd.go:231]: This function is very long, should we break it into smaller functions to improve readability?

File: cmd/create/operatorroles/cmd.go [hunk 4]
```diff
@@ -197,40 +208,253 @@ func run(cmd *cobra.Command, argv []string) error {
...
+		fmt.Println(commands)
+	default:
+		r.Reporter.Errorf("Invalid mode. Allowed values are %s", aws.Modes)
 		os.Exit(1)
 	}
+	return nil
+}
 
-	env, err := ocm.GetEnv()
+func getOperatorRoleArn(prefix string, operator *cmv1.STSOperator, creator *aws.Creator, path string) string {
```
Comment (oriAdler) [inline cmd/create/operatorroles/cmd.go:387]: This function is very similar to `getOperatorRoleArn` in `cmd/create/cluster/cmd.go`.
Can we extract the function and reuse it?

Comment (gdbranco) [reply cmd/create/operatorroles/cmd.go:387]: Will check in on that

**Pull request merged**